tasks/agent.py

```diff
@@ -857,6 +858,132 @@ def _send_build_metrics(ctx, overall_duration):
         print(r.text)
 
 
+def _get_build_images(ctx):
+    # We intentionally include both build images & their test suffixes in the pattern
+    # as a test image and the merged version shouldn't share their cache
+    tags = ctx.run("grep -E 'DATADOG_AGENT_.*BUILDIMAGES' .gitlab-ci.yml | cut -d ':' -f 2", hide='stdout').stdout
+    return (t.strip() for t in tags.splitlines())
+
+
+def _get_environment_for_cache() -> dict:
+    """
+    Compute a hash from the environment after excluding irrelevant/insecure
+    environment variables to ensure we don't omit a variable
+    """
+
+    def env_filter(item):
```

Comment:
Open question, should we move this out of the file? I fear it might change semi regularly especially at first, and it's quite a large chunk of code

Reply:
I agree, I'm moving the whole into a `lib/omnibus_cache.py` file

Reply:
Thanks!